inst/include/population.h

```diff
@@ -136,6 +140,17 @@ namespace smc {
 
         return *this;
     }
+    
+    /// This function performs a stable calculation of the log sum of the weights, which is useful for
+    /// normalising weights, calculating the effective sample size and estimating the normalising constant.
+    ///
+    /// \param logw The log weights of interest.
+    inline double stableLogSumWeights(const arma::vec & logw){
```

Comment:
I'm doing this too quickly, because I need to check out of my accommodation shortly, but is this "code in a header"? With the exception of /templated/ functions which are instructions to the compiler to generate code rather than code itself, or inline member functions, it's usual to keep only declarations in the header (to prevent the kind of compiler error that Dirk encountered) with the declaration living elsewhere in a source file.